spec.bs

```diff
@@ -0,0 +1,401 @@
+1. Return "<code>[=worklet type identifier/shared-storage=]</code>". -->
+
+To <dfn export algorithm for="PrivateAggregation">obtain a reporting endpoint</dfn> given an [=origin=] |reportingOrigin| and [=worklet type identifier=] |api|:
+1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |path| be the [=string/concatenation=] of «"<code>.well-known/private-aggregation/report-</code>", |api|».
+1. Let |base| be the result on running the [=URL parser=] on the [=serialization of an origin|serialization=] of |reportingOrigin|.
+1. Return the result of running the [=URL parser=] on |path| with |base|.
+
+Issue: Remove export when this definition is used.
+
+To <dfn for="PrivateAggregation">obtain an aggregatable report</dfn> given an [=origin=] |reportingOrigin|, a [=worklet type identifier=] |api| and a [=list=] of {{PAHistogramContribution}} |contributions|:
+1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |reportTime| be the result of running [=PrivateAggregation/obtain a report delivery time=].
+1. Let |report| be a new [=aggregatable report=] with the items:
+    : [=aggregatable report/reporting origin=]
+    :: |reportingOrigin|
+    : [=aggregatable report/original report time=]
+    :: |reportTime|
+    : [=aggregatable report/report time=]
+    :: |reportTime|
+    : [=aggregatable report/contributions=]
+    :: |contributions|
+    : [=aggregatable report/api=]
+    :: |api|
+    : [=aggregatable report/report ID=]
+    :: The result of [=generating a random UUID=].
+
+Scheduling reports {#scheduling-reports}
+----------------------------------------
+
+To <dfn algorithm for="PrivateAggregation">obtain a report delivery time</dfn>, run the following steps:
+
+Issue: Fill in
+
+The <dfn algorithm for="PrivateAggregation">report scheduling steps</dfn> given an [=origin=] |reportingOrigin|, a [=worklet type identifier=] |api| and a [=list=] of {{PAHistogramContribution}} |contributions| are as follows:
+1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |truncatedContributions| be a new [=list/is empty|empty=] [=list=].
+1. If |contributions| has a [=list/size=] greater than [=max report contributions=]:
+    1. [=set/For each=] |n| of [=the range=] 0 to [=max report contributions=], exclusive:
+        1. [=set/Append=] |contributions|[|n|] to |truncatedContributions|.
+1. Otherwise, set |truncatedContributions| to |contributions|.
+1. Let |contributionSum| be 0.
+1. [=set/iterate|For each=] |contribution| of |truncatedContributions|:
+    <!-- 1. [=Assert=]: |contribution| is a {{PAHistogramContribution}}. -->
+    1. [=Assert=]: |contribution|'s |value| is non-negative.
+    1. Add |value| to |contributionSum|.
+1. Let |sufficientBudget| be the result of running [=PrivateAggregation/consume budget if permitted=] on |contributionSum|, |reportingOrigin| and |api|.
+1. If not |sufficientBudget|, return.
```

Comment:
Is not.. used elsewhere? Or shall we say "If |sufficientBudget| is false"?

Reply:
Changed to "is false"